packages/now-build-utils/src/fs/node-version.ts

```diff
@@ -42,15 +47,33 @@ export async function getSupportedNodeVersion(
         );
       }
     } else {
-      if (!silent) {
-        throw new Error(
-          'Found `engines` in `package.json` with an unsupported node range: ' +
-            engineRange +
-            '\nPlease use one of the following supported ranges: ' +
-            JSON.stringify(supportedOptions.map(o => o.range))
-        );
-      }
+      throw new Error(
+        'Found `engines` in `package.json` with an unsupported node range: ' +
+          engineRange +
+          '\nPlease use one of the following supported ranges: ' +
+          JSON.stringify(supportedOptions.map(o => o.range))
+      );
     }
   }
+
+  const { range, discontinueDate } = selection;
+  if (discontinueDate && !isDiscontinued(selection)) {
+    const d = discontinueDate.toISOString().split('T')[0];
+    const validRanges = supportedOptions
+      .filter(o => !o.discontinueDate)
+      .map(o => o.range);
+    console.warn(
+      `Warning: Node ${range} will be discontinued on ${d}.\n` +
+        `Deployments created on or after ${d} will fail to build.\n` +
+        'Please use one of the following supported `engines` in `package.json`: ' +
+        JSON.stringify(validRanges)
+    );
+  }
+
   return selection;
 }
+
+function isDiscontinued({ discontinueDate }: NodeVersion): boolean {
+  const today = new Date();
+  return typeof discontinueDate !== 'undefined' && discontinueDate <= today;
```

Comment:
```suggestion
  return discontinueDate !== undefined && discontinueDate <= today;
```

No need to use `typeof` when checking for `undefined`

Reply:
I think it's only useful when the variable wasn't declared

Reply:
Old habits die hard 😇

It used to be "best practice" to use typeof since someone could assign `undefined = 'something'` and it would no longer be undefined 😄

It looks like this was [fixed in ES5](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/undefined) so apparently I've been using JS for too long 🤣👴

Reply:
:D TIL, didn't know it was possible to "overwrite" `undefined`